garden-service/src/plugins/kubernetes/local/kind.ts

```
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2018 Garden Technologies, Inc. <info@garden.io>
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { exec } from "../../../util/util"
+import { kubectl } from "../kubectl"
+import { LogEntry } from "../../../logger/log-entry"
+import { safeLoad } from "js-yaml"
+
+const CLUSTER_NOT_FOUND = "CLUSTER_NOT_FOUND"
+
+export async function loadLocalImage(buildResult: any, config: any, log: LogEntry): Promise<void> {
+  try {
+    const clusterName = await getClusterForContext(config.context)
+    if (clusterName != CLUSTER_NOT_FOUND) {
+      await exec("kind", ["load", "docker-image", buildResult.details.identifier, `--name=${clusterName}`])
+    }
+  } catch (err) {
+    log.error(
+      `An attempt to load the image ${buildResult.details.identifier} into kind cluster with context ${config.context} failed`
```

I think the "with context ..." part is sorta unnecessary here for the error message. Also, we generally prefer to throw a new error in these cases, e.g. `throw new RuntimeError(`An attempt to load image ${buildResult.details.identifier} into the kind cluster failed: ${err.message}`, { err })` (or something like that)

---

done